pennylane/pauli/conversion.py

```diff
@@ -220,7 +129,212 @@ def pauli_decompose(
+    + -0.5 * X(a) @ Z(b)
+    + 1.0 * Y(a) @ Y(b)
+    + -0.5 * Z(a) @ X(b)
+    + -0.5 * Z(a) @ Y(b)
+    """
+    n = int(qml.math.log2(len(H)))
+    N = 2**n
+
+    if H.shape != (N, N):
+        raise ValueError("The matrix should have shape (2**n, 2**n), for any qubit number n>=1")
+
+    if check_hermitian and not qml.math.allclose(H, qml.math.conjugate(qml.math.transpose(H))):
+        raise ValueError("The matrix is not Hermitian")
+
+    coeffs, obs = _generalized_pauli_decompose(
+        H, hide_identity=hide_identity, wire_order=wire_order, pauli=pauli, padding=False
+    )
+    coeffs = qml.math.real(coeffs)
+
+    if pauli:
+        return PauliSentence(
+            {
+                PauliWord({w: o for o, w in obs_n_wires}): coeff
+                for coeff, obs_n_wires in zip(coeffs, obs)
+            }
+        )
+
+    with qml.QueuingManager.stop_recording():
+        return Hamiltonian(coeffs, obs)
+
+
+def pauli_decompose_with_phase(
+    matrix,
+    hide_identity=False,
+    wire_order=None,
+    pauli=False,
+    padding=False,
+) -> Union[Tuple[qml.math.array, list], PauliSentence]:
+    r"""Decomposes any matrix into a linear combination of Pauli operators.
+
+    This method converts any matrix to a weighted sum of Pauli words acting on :math:`n` qubits
+    in time :math:`O(n 4^n)`. The input matrix is first padded with zeros if its dimensions are not
+    :math:`2^n\times 2^n` and written as a quantum state in the computational basis following the
+    `channel-state duality <https://en.wikipedia.org/wiki/Channel-state_duality>`_.
+    A Bell basis transformation is then performed using the
+    `Walsh-Hadamard transform <https://en.wikipedia.org/wiki/Hadamard_transform>`_, after which
+    coefficients for each of the :math:`4^n` Pauli words are computed while accounting for the
+    phase from each ``PauliY`` term occurring in the word.
+
+    Args:
+        matrix (tensor_like[complex]): any matrix M, the keyword argument ``padding=True``
+            should be provided if the dimension of M is not :math:`2^n\times 2^n`.
+        hide_identity (bool): does not include the Identity observable within
+            the tensor products of the decomposition if ``True``.
+        wire_order (list[Union[int, str]]): the ordered list of wires with respect
+            to which the operator is represented as a matrix.
+        pauli (bool): return a PauliSentence instance if ``True``.
+        padding (bool): makes the function compatible with rectangular matrices and square matrices
+            that are not of shape :math:`2^n\times 2^n` by padding them with zeros if ``True``.
+
+    Returns:
+        Union[Tuple[tensor_like[complex], list[~.Operation]], ~.PauliSentence]: the matrix decomposed as
+        a linear combination of Pauli operators as as a ``tuple`` consisting of an ``array`` of complex
+        coefficients and a ``list`` of corresponding Pauli terms, or :class:`~.PauliSentence` instance.
```

Comment:
I personally don't mind that this returns a different type from `pauli_decompose`, and it's clearly documented, but it might be a bit unexpected for users who want to try out each of the functions. I'd have expected that they work the same way except that `pauli_decompose` doesn't have a phase 😄 could you perhaps elaborate a bit on the motivation for this difference?